### ISSUE TITLE
Agent starting with wrong configuration (MANAGER_IP)

### ISSUE THREAD

**Issue by jmv74211** [wazuh/wazuh#3789]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 3.9.3-3929 | All | Agent | Packages | Debian9 |

Hello team,

When installing the wazuh-agent, I noticed that the agent has started correctly without specifying the IP address of the manager. Its possible cause could be that the text MANAGER_IP is detected as a hostname, and therefore, it does not produce any error saying that it is necessary to specify this configuration. We have been talking about that and we considerate that is a not desirable behaviour, because the agent seems to be correctly running, but actually it could not connect to the manager. We considerate that the agent should not start if there is not a correct manager ip address or at least, register it in logs.

```
2019/08/01 08:45:18 agent-auth: INFO: Started (pid: 2635).
2019/08/01 08:45:18 agent-auth: INFO: No authentication password provided.
2019/08/01 08:45:18 agent-auth: INFO: Connected to 172.16.1.17:1515
2019/08/01 08:45:18 agent-auth: INFO: Using agent name as: debian9
2019/08/01 08:45:18 agent-auth: INFO: Send request to manager. Waiting for reply.
2019/08/01 08:45:18 agent-auth: INFO: Received response with agent key
2019/08/01 08:45:18 agent-auth: INFO: Valid key created. Finished.
2019/08/01 08:45:18 agent-auth: INFO: Connection closed.
2019/08/01 08:46:20 ossec-execd: INFO: Started (pid: 2669).
2019/08/01 08:46:26 ossec-agentd: INFO: Using notify time: 10 and max time to reconnect: 60
2019/08/01 08:46:26 ossec-agentd: INFO: Version detected -> Linux |debian9 |4.9.0-9-amd64 |#1 SMP Debian 4.9.168-1+deb9u2 (2019-05-13) |x86_64 [Debian GNU/Linux|debian: 9 (stretch)] - Wazuh v3.9.3
2019/08/01 08:46:26 ossec-agentd: INFO: (1410): Reading authentication keys file.
2019/08/01 08:46:26 ossec-agentd: INFO: Using AES as encryption method.
2019/08/01 08:46:26 ossec-agentd: INFO: Started (pid: 2674).
2019/08/01 08:46:26 ossec-agentd: INFO: Server IP Address: MANAGER_IP
2019/08/01 08:46:26 ossec-logcollector: INFO: Monitoring output of command(360): df -P
2019/08/01 08:46:26 ossec-logcollector: INFO: Monitoring full output of command(360): netstat -tulpn | sed 's/\([[:alnum:]]\+\)\ \+[[:digit:]]\+\ \+[[:digit:]]\+\ \+\(.*\):\([[:digit:]]*\)\ \+\([0-9\.\:\*]\+\).\+\ \([[:digit:]]*\/[[:alnum:]\-]*\).*/\1 \2 == \3 == \4 \5/' | sort -k 4 -g | sed 's/ == \(.*\) ==/:\1/' | sed 1,2d
2019/08/01 08:46:26 ossec-logcollector: INFO: Monitoring full output of command(360): last -n 20
2019/08/01 08:46:26 ossec-logcollector: INFO: (1950): Analyzing file: '/var/ossec/logs/active-responses.log'.
2019/08/01 08:46:26 ossec-logcollector: INFO: (1950): Analyzing file: '/var/log/messages'.
2019/08/01 08:46:26 ossec-logcollector: INFO: (1950): Analyzing file: '/var/log/auth.log'.
2019/08/01 08:46:26 ossec-logcollector: INFO: (1950): Analyzing file: '/var/log/syslog'.
2019/08/01 08:46:26 ossec-logcollector: INFO: (1950): Analyzing file: '/var/log/dpkg.log'.
2019/08/01 08:46:26 ossec-logcollector: INFO: (1950): Analyzing file: '/var/log/kern.log'.
2019/08/01 08:46:26 ossec-logcollector: INFO: Started (pid: 2686).
2019/08/01 08:46:26 wazuh-modulesd: INFO: Process started.
2019/08/01 08:46:26 wazuh-modulesd:control: INFO: Starting control thread.
2019/08/01 08:46:26 sca: INFO: Module started.
2019/08/01 08:46:26 sca: INFO: Starting Security Configuration Assessment scan.
2019/08/01 08:46:26 sca: INFO: Starting evaluation of policy: 'cis_debian_linux_rcl.yml'
2019/08/01 08:46:26 wazuh-modulesd:syscollector: INFO: Module started.
2019/08/01 08:46:26 wazuh-modulesd:osquery: INFO: Module disabled. Exiting...
2019/08/01 08:46:26 wazuh-modulesd:ciscat: INFO: Module disabled. Exiting...
2019/08/01 08:46:26 wazuh-modulesd:oscap: INFO: Module disabled. Exiting...
2019/08/01 08:46:27 wazuh-modulesd:syscollector: INFO: Starting evaluation.
2019/08/01 08:46:29 ossec-syscheckd: INFO: Started (pid: 2680).
2019/08/01 08:46:29 ossec-syscheckd: INFO: (6003): Monitoring directory: '/etc', with options 'perm | size | owner | group | md5sum | sha1sum | sha256sum | mtime | inode'.
2019/08/01 08:46:29 ossec-syscheckd: INFO: (6003): Monitoring directory: '/usr/bin', with options 'perm | size | owner | group | md5sum | sha1sum | sha256sum | mtime | inode'.
2019/08/01 08:46:29 ossec-syscheckd: INFO: (6003): Monitoring directory: '/usr/sbin', with options 'perm | size | owner | group | md5sum | sha1sum | sha256sum | mtime | inode'.
2019/08/01 08:46:29 ossec-syscheckd: INFO: (6003): Monitoring directory: '/bin', with options 'perm | size | owner | group | md5sum | sha1sum | sha256sum | mtime | inode'.
2019/08/01 08:46:29 ossec-syscheckd: INFO: (6003): Monitoring directory: '/sbin', with options 'perm | size | owner | group | md5sum | sha1sum | sha256sum | mtime | inode'.
2019/08/01 08:46:29 ossec-syscheckd: INFO: (6003): Monitoring directory: '/boot', with options 'perm | size | owner | group | md5sum | sha1sum | sha256sum | mtime | inode'.
2019/08/01 08:46:29 ossec-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/mtab'
2019/08/01 08:46:29 ossec-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/hosts.deny'
2019/08/01 08:46:29 ossec-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/mail/statistics'
2019/08/01 08:46:29 ossec-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/random-seed'
2019/08/01 08:46:29 ossec-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/random.seed'
2019/08/01 08:46:29 ossec-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/adjtime'
2019/08/01 08:46:29 ossec-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/httpd/logs'
2019/08/01 08:46:29 ossec-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/utmpx'
2019/08/01 08:46:29 ossec-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/wtmpx'
2019/08/01 08:46:29 ossec-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/cups/certs'
2019/08/01 08:46:29 ossec-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/dumpdates'
2019/08/01 08:46:29 ossec-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/svc/volatile'
2019/08/01 08:46:29 ossec-syscheckd: INFO: (6206): Ignore 'file' entry '/sys/kernel/security'
2019/08/01 08:46:29 ossec-syscheckd: INFO: (6206): Ignore 'file' entry '/sys/kernel/debug'
2019/08/01 08:46:29 ossec-syscheckd: INFO: (6206): Ignore 'file' entry '/dev/core'
2019/08/01 08:46:29 ossec-syscheckd: INFO: (6207): Ignore 'file' sregex '^/proc'
2019/08/01 08:46:29 ossec-syscheckd: INFO: (6207): Ignore 'file' sregex '.log$|.swp$'
2019/08/01 08:46:29 ossec-syscheckd: INFO: (6004): No diff for file: '/etc/ssl/private.key'
2019/08/01 08:46:29 rootcheck: INFO: Started (pid: 2680).
2019/08/01 08:46:44 ossec-syscheckd: INFO: (6010): File integrity monitoring scan frequency: 43200 seconds
2019/08/01 08:46:44 ossec-syscheckd: INFO: (6008): File integrity monitoring scan started.
2019/08/01 08:54:11 wazuh-modulesd:syscollector: INFO: Module finished.
```

I opened this issue to notify that behaviour, if that is the expected one, please close this issue.

## Reproduction steps

- Use [this Vagrantfile](https://github.com/wazuh/wazuh-tools/blob/master/vagrant/cicd_team/Vagrantfile)  to create the Debian9 machine.

- Install the wazuh agent as it indicates in this [documentation](https://documentation.wazuh.com/3.9/installation-guide/installing-wazuh-agent/wazuh_agent_linux_deb.html) without automating the agent registration.

- Register the agent using: `/var/ossec/bin/agent-auth -m 172.xx.xx.xx`

**Comment by MiguelCasaresRobles** [comment]
This behavior is happening in Centos7 as well, and without specifying the IP address. 

For example:

```
   <server>
      <address></address>
      <port>1514</port>
      <protocol>udp</protocol>
    </server> 
```

This configuration will work.

**Comment by chemamartinez** [comment]
Merged #3951
